src/AppForeach.Framework.EntityFrameworkCore/DataType/EntityTypeBuilderExtensions.cs

```diff
@@ -1,9 +1,51 @@
-﻿using Microsoft.EntityFrameworkCore.Metadata.Builders;
+﻿using AppForeach.Framework.DataType;
+using AppForeach.Framework.DataType.Facets;
+using AppForeach.Framework.DependencyInjection;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+using System.Linq;
 
 namespace AppForeach.Framework.EntityFrameworkCore.DataType
 {
     public static class EntityTypeBuilderExtensions
     {
-        public static void FromEntitySpecification<T>(this EntityTypeBuilder<T> ext) where T : class { }
+        public static void FromEntitySpecification<T>(this EntityTypeBuilder<T> builder) where T : class
+        {
+            var assemblyToScan = typeof(T).Assembly;
```

Comment:
@viktor-kulkov  here is the new approach based on assembly scanning. I also went ahead and removed DI/Service Locator PoC code